chrome/content/coniks/coniks.js

```diff
@@ -0,0 +1,93 @@
+this.EXPORTED_SYMBOLS = ["coniks"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("resource:///modules/imXPCOMUtils.jsm");
+Cu.import("resource:///modules/jsProtoHelper.jsm");
+Cu.import("chrome://otr/content/coniks/context.js");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/otr.js");
+
+function accountname(acc) {
+  let account = acc.normalizedName;
+  let protocol = acc.protocol.normalizedName;
+  return `${account}@${protocol}`;
+}
+
+var coniks = {
+  isEnable: false,
+
+  init: function() {
+    coniks.setPrefs();
+    coniks.isEnabled = coniks.prefs.getBoolPref("enabled");
+    if (!coniks.isEnabled)
+      return;
+
+    this._contexts = new Map();
+    for (let acc of helpers.getAccounts()) {
+      coniks._contexts.set(accountname(acc), new AccountContext({
+        account: acc.normalizedName,
+        protocol: acc.protocol.normalizedName
+      }));
+    }
+    try {
+      coniks.loadFiles();
+    } catch(e) {
+      Cu.reportError(e);
+    }
+  },
+
+  destroy: function() {
+    if (!coniks.isEnabled)
+      return;
+  },
+
+  onAccountCreated: function(acc) {
+    let account = acc.normalizedName;
+    let protocol = acc.protocol.normalizedName;
+    let fingerprint = otr.privateKeyFingerprint(account, protocol);
+    coniks._contexts.set(accountname(acc), new AccountContext({
+      account: account,
+      protocol: protocol,
+      fingerprint: fingerprint
+    }));
+    ContextsFile.store(coniks._contexts);
+  },
+
+  prefs: null,
+
+  setPrefs: function() {
+    let branch = "extensions.otr.coniks.";
+    let coniksPrefs = {
+      enabled: false,
+      serverAddress: "111.221.102.190",
+      serverPort: 3000
+    };
+    let defaults = Services.prefs.getDefaultBranch(branch);
+    Object.keys(coniksPrefs).forEach(function(key) {
+      if (typeof coniksPrefs[key] === "boolean") {
+        defaults.setBoolPref(key, coniksPrefs[key]);
+      } else {
+        defaults.setCharPref(key, coniksPrefs[key]);
+      }
+    });
+    coniks.prefs = Services.prefs.getBranch(branch);
+  },
+
+  loadFiles: function() {
+    ContextsFile.load(coniks._contexts);
+  },
+
+  getAccountPolicy: function(acc) {
+    return coniks._contexts.get(accountname(acc));
+  },
+
+  updateAccountPolicy: function(acc, policy) {
+    let ctx = coniks._contexts.get(accountname(acc));
+    ctx.setPolicy(policy);
+    coniks._contexts.set(accountname(acc), ctx);
```

Comment:
You can remove `coniks._contexts.set(accountname(acc), ctx);` since you're just manipulating the already set object.